Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp

```diff
@@ -1557,7 +1557,11 @@ std::optional<unsigned> gstGetAutoplugSelectResult(ASCIILiteral nick)
 #if !GST_CHECK_VERSION(1, 20, 0)
 GstBuffer* gst_buffer_new_memdup(gconstpointer data, gsize size)
 {
+#if GLIB_CHECK_VERSION(2, 68, 0)
     gpointer copiedData = g_memdup2(data, size);
+#else
+    gpointer copiedData = g_memdup(data, size);
```

Comment:
This is an integer truncation vulnerability (which is why `g_memdup()` is deprecated). Now if `size > UINT_MAX` then GstBuffer thinks it has a larger buffer than it really does.

I would just open code `g_malloc()` then `memcpy()`. Or you could truncate safely by doing `size = static_cast<unsigned>(size)`.

Reply:
let's do that truncation then.